extras/semantics-toAST.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var pexprs = require('../src/pexprs');
+var MatchResult = require('../src/MatchResult');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+function toAST(res, mapping) {
```

Comment:
It would still be nice to have at least a couple of sentences of documentation here, to describe what this function does and how it works. I still don't fully understand exactly how the mapping works.